### PR TITLE
Refactor iothub_endpoint_* resources to use id parser

### PR DESCRIPTION
**Pull request by stephybun**

### CODE REVIEW

File: internal/services/iothub/iothub_endpoint_eventhub_resource.go [hunk 1]
```diff
@@ -153,14 +154,14 @@ func resourceIotHubEndpointEventHubRead(d *pluginsdk.ResourceData, meta interfac
...
 	if err != nil {
 		return err
 	}
 
-	resourceGroup := parsedIothubEndpointId.ResourceGroup
-	iothubName := parsedIothubEndpointId.Path["IotHubs"]
-	endpointName := parsedIothubEndpointId.Path["Endpoints"]
+	resourceGroup := id.ResourceGroup
+	iothubName := id.IotHubName
+	endpointName := id.EndpointName
```
Comment (katbyte) [inline internal/services/iothub/iothub_endpoint_eventhub_resource.go:164]: we don't need to assign these to variables anymore
```suggestion
```

File: internal/services/iothub/iothub_endpoint_eventhub_resource.go [hunk 2]
```diff
@@ -193,14 +194,14 @@ func resourceIotHubEndpointEventHubDelete(d *pluginsdk.ResourceData, meta interf
...
```
Comment (katbyte) [inline internal/services/iothub/iothub_endpoint_eventhub_resource.go:204]: and here
```suggestion
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @stephybun - overall looks good but theres some unnecessary variables we can remove now that we have the id object. i only commented on a couple out of all the instances

**Comment by stephybun** [comment]
I removed those var assignments, also from the iothub_dps_* resources which I refactored and merged yesterday in #13304

**Review by katbyte — APPROVED** [review]
Thanks @stephybun! LGTM 🏗️

**Pull request merged**